src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -14635,29 +14706,615 @@ public S(A a, B b)
             comp.VerifyDiagnostics(
-                // (7,26): warning CS8604: Possible null reference argument for parameter 'a' in 'S.S(A a, B b)'.
+                // (7,26): error CS8600: Cannot convert null to non-nullable reference.
                 //     public S(B b) : this(null, b)
-                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "null").WithArguments("a", "S.S(A a, B b)").WithLocation(7, 26));
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(7, 26));
+        }
+
+        // PROTOTYPE(NullableReferenceTypes): Update other tests with WithNullCheckingFeature(flags: NullableReferenceFlags.None) to verify expected changes.
+
+        [Fact]
+        public void WarningOnConversion_Assignment()
+        {
+            var source =
+@"class Person
+{
+    internal string FirstName { get; set; }
+    internal string LastName { get; set; }
+    internal string? MiddleName { get; set; }
+}
+class Program
+{
+    static void F(Person p)
+    {
+        p.LastName = null;
+        p.LastName = (string)null;
+        p.LastName = (string?)null;
+        p.LastName = null as string;
+        p.LastName = null as string?;
+        p.LastName = default(string);
+        p.LastName = default;
+        p.FirstName = p.MiddleName;
+        p.LastName = p.MiddleName ?? null;
+    }
+}";
+
+            var comp = CreateStandardCompilation(
+                new[] { source, attributesDefinitions },
+                parseOptions: TestOptions.Regular.WithNullCheckingFeature(flags: NullableReferenceFlags.None).WithLanguageVersion(LanguageVersion.CSharp7_1));
+            comp.VerifyDiagnostics(
+                // (14,22): warning CS8601: Possible null reference assignment.
+                //         p.LastName = null as string;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "null as string").WithLocation(14, 22),
+                // (15,22): warning CS8601: Possible null reference assignment.
+                //         p.LastName = null as string?;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "null as string?").WithLocation(15, 22),
+                // (18,23): warning CS8601: Possible null reference assignment.
+                //         p.FirstName = p.MiddleName;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "p.MiddleName").WithLocation(18, 23),
+                // (19,22): warning CS8601: Possible null reference assignment.
+                //         p.LastName = p.MiddleName ?? null;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "p.MiddleName ?? null").WithLocation(19, 22));
+
+            comp = CreateStandardCompilation(
+                new[] { source, attributesDefinitions },
+                parseOptions: TestOptions.Regular.WithNullCheckingFeature(flags: NullableReferenceFlags.IncludeNonNullWarnings).WithLanguageVersion(LanguageVersion.CSharp7_1));
```

Comment:
I guess, the fact that the baseline says **error** instead of **warning** confused me:
```
                // (11,22): error CS8600: Cannot convert null to non-nullable reference.
                //         p.LastName = null;
                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(11, 22),
```

Why is this happening for this warning?